### PR TITLE
Fix/fastify swagger bump

### PR DESCRIPTION
**Pull request by pedropalhari**
This branch aims to bring `@fastify/swagger` into its latest version, going through a breaking change on `@^8` that splits the package into `@fastify/swagger` and `@fastify/swagger-ui`.

Also, adds `zod` as a `devDependency` (required for testing), fixes a missing interop on the tsconfig and fixes the example on README.md.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "compilerOptions": {
     "declaration": true,
     "rootDir": "src",
-    "outDir": "dist"
+    "outDir": "dist",
+    "esModuleInterop": true
```
Comment (kibertoad) [inline tsconfig.json:6]: why was this necessary?

File: README.md [hunk 2]
```diff
@@ -86,5 +94,15 @@ app.after(() => {
   });
 });
 
-await app.ready();
+async function run() {
```
Comment (kibertoad) [inline README.md:97]: why this change?

File: tsconfig.json (hunk 1 above)
Comment (pedropalhari) [reply tsconfig.json:6]: Hi @kibertoad!

Due to the usage of both "import" and "require" on the test `fastify-swagger.spec.ts`, the following warning pops up multiple times:

```
ts-jest[config] (WARN) message TS151001: If you have issues related to imports, you should consider setting `esModuleInterop` to `true` in your TypeScript configuration file (usually `tsconfig.json`). See https://blogs.msdn.microsoft.com/typescript/2018/01/31/announcing-typescript-2-7/#easier-ecmascript-module-interoperability for more information.
```

I just followed the advice and added the required property from tsconfig.json. Another way to solve this would be changing the "require" to an import but I didn't dig into that.

File: README.md (hunk 2 above)
Comment (pedropalhari) [reply README.md:97]: I made that change because the first example in the README, `How to use?`, is compatible to both CommonJS and ES modules projects while the second one, as it uses a top-level await, is only available to ES modules projects.

With this now users can just copy-paste the second `How to use together with @fastify/swagger` example without bothering to adapt it to ES modules (just like the first one).

**Review by kibertoad — APPROVED** [review]

**Comment by pedropalhari** [comment]
Going to fix the lint issue, I wasn't attentive enough to catch it before pushing the branch, my bad.

**Comment by pedropalhari** [comment]
Done @turkerdev! I rewrote the import to be a common `import * as ... from "..."`

File: tsconfig.json (hunk 1 above)
Comment (kibertoad) [reply tsconfig.json:6]: is this still needed after the import change?

**Review by turkerdev — APPROVED** [review]
Thank you
